src/dwarf/Gparser.c

```diff
@@ -440,8 +440,13 @@ fetch_proc_info (struct dwarf_cursor *c, unw_word_t ip)
      continue, and it's important we get this right, as 'ip' could be
      right at the function entry and hence FDE edge, or at instruction
      that manipulates CFA (push/pop). */
-  if (c->use_prev_instr)
+  if (c->use_prev_instr) {
+#if defined(__arm__)
```

Comment:
afaik this won't work when unwinding an arm stack copy on a x86 machine, or? this is required for perf record on arm followed by perf report on x86 e.g.

i.e. instead of a defined, this need to check the source ABI somehow?

Reply:
Good to know! thanks @milianw 
I will clean up the thumb flag in other place where only effect arm device

Reply:
Generally, when the libunwind is built for specific target architecture, the UNW_TARGET_<something> is defined.  Look into the source for its appearance to see how to use it.

That said (and this is probably a generic note for most load/store arches), wouldn't it be better to decrement by 4, and by 2 for thumb mode, on arm ?

Reply:
@kostikbel yup, minus 2/4 look better to me too. But `-1` looks like is not enough for all the arch, but it actually move the `$IP` to the right address range. It's kinda of enough. I'm quite neutral on this.